### PR TITLE
How do use `cast` to relay messages

### PR DESCRIPTION
**Pull request by qbzzt**
**Description**

A separate tutorial because this is a rare edge case.

**Tests**

N/A

**Additional context**

N/A

**Metadata**

N/A

### CODE REVIEW

File: pages/interop/tutorials/message-passing/relay-with-cast.mdx
Comment (tremarkley) [inline pages/interop/tutorials/message-passing/relay-with-cast.mdx:1]: should this be deduplicated with https://github.com/ethereum-optimism/docs/blob/main/pages/interop/tutorials/relay-messages-cast.mdx?

File: pages/interop/tutorials/message-passing/relay-with-cast.mdx [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+```
+
+To solve some [denial of service issues](https://github.com/ethereum-optimism/design-docs/blob/main/protocol/interop-access-list.md), relay transactions have to have an access list with a checksum of the message.
+This lets sequencers know what executing messages to expect in a transaction, which makes it easy not to include transactions that are invalid because they rely on messages that were never sent.
+
+The [algorithm to calculate the access list](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/src/L2/CrossL2Inbox.sol#L87-L115) is a bit complicated, but you don't need to worry about it.
+Supersim exposes [RPC calls](https://supersim.pages.dev/guides/interop/cast?highlight=manuall#7-construct-the-access-list-for-the-message) that calculates it for you on port 8420.
+Even if you use a different interop cluster that lost autorelay for some reason, the code above should calculate the correct access list because this is a [pure function](https://en.wikipedia.org/wiki/Pure_function). The other RPC call, `admin_getAccessListByMsgHash` is not and therefore is less flexible.
```
Comment (krofax) [inline pages/interop/tutorials/message-passing/relay-with-cast.mdx:192]: This line is confusing, can we update this text to something like this.

"The code above will calculate the correct access list even if you're using a different interop cluster where autorelay is not functioning. This is because the code implements a [pure function](https://en.wikipedia.org/wiki/Pure_function), which produces consistent results regardless of external state. In contrast, the `admin_getAccessListByMsgHash` RPC call is not a pure function, making it more dependent on system state and therefore less flexible in these situations."?

File: pages/interop/tutorials/message-passing/relay-with-cast.mdx [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+cast call $GREETER_B_ADDRESS "greet()(string)" --rpc-url $URL_CHAIN_B
+```
+
+Again, show the current greeting.
+Now it's the new one.
+
+## Next steps
+
+*   Review the [Superchain Interop Explainer](/interop/explainer) for answers to common questions about interoperability.
+*   Read the [Message Passing Explainer](/interop/message-passing) to understand what happens "under the hood".
```
Comment (krofax) [inline pages/interop/tutorials/message-passing/relay-with-cast.mdx:219]:
```suggestion
*   Review the [Superchain interop explainer](/interop/explainer) for answers to common questions about interoperability.
*   Read the [message passing explainer](/interop/message-passing) to understand what happens "under the hood".
```

**Review by krofax — CHANGES_REQUESTED** [review]

**Review by krofax — CHANGES_REQUESTED** [review]
@qbzzt added some comments.

File: public/tutorials/setup-for-manual-relay.sh [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+rm -rf manual-relay
+mkdir manual-relay
+cd manual-relay
+
+PRIVATE_KEY=0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80
+USER_ADDRESS=`cast wallet address --private-key $PRIVATE_KEY`
+URL_CHAIN_A=http://localhost:9545
+URL_CHAIN_B=http://localhost:9546
+
```
[the remark this answers is not in the document]
Comment (qbzzt) [reply public/tutorials/setup-for-manual-relay.sh:11]: This is a well known private key used only for testing. No need to keep it a secret or complicate matters.

File: pages/interop/tutorials/message-passing/relay-with-cast.mdx (hunk 1 above)
Comment (qbzzt) [reply pages/interop/tutorials/message-passing/relay-with-cast.mdx:193]: That is better. Except, "more dependent" doesn't work great, "dependent" is better.

File: public/tutorials/setup-for-manual-relay.sh [hunk 4]
```diff
@@ -0,0 +1,152 @@
+#! /bin/sh
+
+rm -rf manual-relay
+mkdir manual-relay
+cd manual-relay
+
```
[the remark this answers is not in the document]
Comment (qbzzt) [reply public/tutorials/setup-for-manual-relay.sh:6]: Not needed, we just created the directory.

**Review by krofax — APPROVED** [review]
